src/clientapi/debugging/traversal.js

```diff
@@ -0,0 +1,307 @@
+/**
+ * @license
+ * This file is part of the Game Closure SDK.
+ *
+ * The Game Closure SDK is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public License v. 2.0 as published by Mozilla.
+
+ * The Game Closure SDK is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public License v. 2.0 for more details.
+
+ * You should have received a copy of the Mozilla Public License v. 2.0
+ * along with the Game Closure SDK.  If not, see <http://mozilla.org/MPL/2.0/>.
+ */
+
+/*
+ * debug tools for traversing views from the JS console.
+ * This class ends up being in the global scope:
+ *   GLOBAL._DEBUG = new exports();
+ */
+
+/* globals GC, bind, logger */
+
+exports.traverse = function (f) { return GC.app && exports.traverseView(f, GC.app.view); };
+exports.traverseView = function (f, view) {
+	var data = f(view);
+	var subviews = view.getSubviews().map(bind(this, 'traverseView', f));
+	return {
+		uid: view.uid,
+		data: data,
+		subviews: subviews.length ? subviews : undefined
+	};
```

Comment:
You changed a bunch of other code to 2 space indents. Any reason this one wasn't included?